File: backend/app/ai/learning_modes.py
"""
Learning Assistant - exactly 6 modes as per Phase 1 spec.
Prompt routing - no extra modes.
"""

LEARNING_MODES = {
    "explain_simply": {
        "label": "Explain Simply",
        "directive": "MODE [EXPLAIN SIMPLY]: Provide a crystal-clear, beginner-friendly explanation. Use intuitive everyday analogies, eliminate opaque jargon where possible, and break down the concepts into easily digestible steps.",
    },
    "explain_detail": {
        "label": "Explain in Detail",
        "directive": "MODE [EXPLAIN IN DETAIL]: Provide an exhaustive, technically rigorous engineering breakdown. Detail underlying architectures, hardware/software interfaces, registers, timing constraints, and memory layouts.",
    },
    "give_example": {
        "label": "Give Example",
        "directive": "MODE [GIVE EXAMPLE]: Provide practical, concrete engineering examples. Include formatted C/C++ firmware snippets, Verilog HDL code, protocol packet hex dumps, or circuit schematics where applicable.",
    },
    "summarize": {
        "label": "Summarize",
        "directive": "MODE [SUMMARIZE]: Provide a condensed, high-yield summary. Use clean bullet points, key takeaways, and critical interview/exam points.",
    },
    "practical_app": {
        "label": "Give Practical Application",
        "directive": "MODE [PRACTICAL APPLICATION]: Emphasize real-world industrial use cases. Explain how companies like Bosch, Qualcomm, Texas Instruments, Intel, and Continental deploy this in automotive ECUs, IoT devices, or chip verification.",
    },
    "quiz_me": {
        "label": "Quiz Me",
        "directive": "MODE [QUIZ ME]: Generate 2-3 engaging multiple-choice technical questions with 4 options (A, B, C, D) each. At the end, include a clear collapsible or marked section with the correct answers and in-depth explanations so the learner can test themselves.",
    },
}

VALID_MODES = set(LEARNING_MODES.keys())


def get_mode_directive(mode: str) -> str:
    return LEARNING_MODES.get(mode, LEARNING_MODES["explain_simply"])["directive"]


def validate_mode(mode: str) -> bool:
    return mode in VALID_MODES

File: src/data/learningModes.ts
import { LearningModeMeta } from '../types';

export const LEARNING_MODES: LearningModeMeta[] = [
  {
    id: 'explain_simply',
    label: 'Explain Simply',
    shortLabel: 'Simple',
    description: 'Simplified explanation suitable for beginners without heavy jargon.',
    icon: 'Sparkles',
    systemDirective: 'Provide a clear, intuitive, and beginner-friendly explanation. Use plain language, accessible analogies, and break down complex engineering concepts into easy-to-digest steps without overwhelming jargon.'
  },
  {
    id: 'explain_detail',
    label: 'Explain in Detail',
    shortLabel: 'Deep Dive',
    description: 'Comprehensive, in-depth engineering breakdown with architecture & mechanisms.',
    icon: 'BookOpen',
    systemDirective: 'Provide an exhaustive, technically rigorous engineering explanation. Detail underlying architectures, hardware/software interfaces, timing diagrams, memory layouts, register-level specifics, and edge cases.'
  },
  {
    id: 'give_example',
    label: 'Give Example',
    shortLabel: 'Examples',
    description: 'Concrete code, circuit, or firmware examples to illustrate concepts.',
    icon: 'Code2',
    systemDirective: 'Illustrate the concept with concrete, practical examples, such as C/C++ embedded code snippets, register configurations, Verilog modules, or state machine flowcharts.'
  },
  {
    id: 'summarize',
    label: 'Summarize',
    shortLabel: 'Summary',
    description: 'Condensed executive summary and key takeaway bullet points.',
    icon: 'ListFilter',
    systemDirective: 'Deliver a concise, high-density summary. Highlight key definitions, core architectural blocks, critical trade-offs, and exam/interview takeaways in clear bullet points.'
  },
  {
    id: 'practical_app',
    label: 'Practical Application',
    shortLabel: 'Real World',
    description: 'Real-world industrial use cases, automotive/medical/aerospace applications.',
    icon: 'Cpu',
    systemDirective: 'Focus on industrial and real-world deployment. Explain how companies (like Bosch, Qualcomm, Texas Instruments, Intel) implement this in automotive ECUs, IoT nodes, avionics, robotics, or consumer hardware.'
  },
  {
    id: 'quiz_me',
    label: 'Quiz Me',
    shortLabel: 'Quiz',
    description: 'Generate interactive multiple-choice & technical assessment questions.',
    icon: 'HelpCircle',
    systemDirective: 'Generate 2-3 engaging technical multiple-choice questions (MCQs) to test understanding of this topic. Clearly number each question (Q1, Q2...), provide 4 distinct options (A, B, C, D), and include a detailed explanation with the correct answer hidden under a "Correct Answer & Explanation" section.'
  }
];
